core/templates/pages/contributor-dashboard-page/opportunities-list/opportunities-list.component.ts

```diff
@@ -20,17 +20,98 @@ require(
   'pages/contributor-dashboard-page/opportunities-list-item/' +
   'opportunities-list-item.component.ts');
 
+import { Subscription } from 'rxjs';
+
 angular.module('oppia').component('opportunitiesList', {
   bindings: {
-    opportunityDataIsLoading: '=',
-    getOpportunities: '&opportunities',
-    isMoreOpportunitiesAvailable: '&moreAvailable',
-    isLabelRequired: '&labelRequired',
-    isProgressBarRequired: '&progressBarRequired',
-    onLoadMoreOpportunities: '&',
-    onClickActionButton: '=',
-    getOpportunityHeadingTruncationLength:
-      '&opportunityHeadingTruncationLength'
+    loadOpportunities: '<',
+    labelRequired: '<',
+    progressBarRequired: '<',
+    loadMoreOpportunities: '<',
+    onClickActionButton: '<',
+    opportunityHeadingTruncationLength: '<'
   },
-  template: require('./opportunities-list.component.html')
+  template: require('./opportunities-list.component.html'),
+  controller: [
+    '$rootScope', '$scope', 'TranslationLanguageService',
+    'OPPORTUNITIES_PAGE_SIZE', function(
+        $rootScope, $scope, TranslationLanguageService,
+        OPPORTUNITIES_PAGE_SIZE) {
+      var ctrl = this;
+      var opportunities = [];
+      ctrl.visibleOpportunities = [];
+      ctrl.directiveSubscriptions = new Subscription();
+
+      ctrl.directiveSubscriptions.add(
+        TranslationLanguageService.onActiveLanguageChanged.subscribe(
+          () => ctrl.$onInit()));
+
+      if ($scope.$parent.$ctrl.hasOwnProperty(
+        'reloadOpportunitiesEventEmitter')) {
+        ctrl.directiveSubscriptions.add(
+          $scope.$parent.$ctrl.reloadOpportunitiesEventEmitter.subscribe(
```

Comment:
I'll create a separate service to have these listeners once the angular team confirms that this can be easily migrated to angular.

/cc @srijanreddy98

Reply:
This can be migrated to angular by moving event emitter into a service and calling the emit function from the service and subscribing here (As we have already done). I would strongly advise against accessing parent scope here or anywhere else. Components are supposed to be independent but using the parent scope implies that it relying on parent in ways that it shouldn't be. The only thing connecting the component with its parent should be bindToController/ bindings. We can consider them as the official API of the component.